### PR TITLE
Remove forgotten deprecations

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | There was some code marked as `To be removed in 1.8`. I removed these deprecations, fix few lines in tests that used this code.
| Type?             | refacto
| Category?         | CO
| BC breaks?        | yes, but people had long time to fix it
| Deprecations?     | no
| Fixed ticket?     | -
| Related PRs       | -
| How to test?      | No need.
| Possible impacts? | 3rd party code that used deprecated functions for a long time. This code was not used anywhere in the core and won't affect anything.

### CODE REVIEW

**Comment by kpodemski** [comment]
@Hlavtox 

As `develop` is 8.1.0, we might need to park this PR for a while. We can remove it in 9.0.

**Comment by Hlavtox** [comment]
@kpodemski Well, yes... but... this code was marked to be deleted in 8.0 and it was only forgotten because of wrong version number in the comment. So I guess > their problem if they use it?

**Comment by kpodemski** [comment]
@Hlavtox 

Unfortunately, it is our problem that we forgot to remove that. We promised not to break stuff in minor versions:
https://github.com/PrestaShop/ADR/blob/master/0017-backward-compatibility-promise.md

**Comment by matks** [comment]
> @Hlavtox
> 
> Unfortunately, it is our problem that we forgot to remove that. We promised not to break stuff in minor versions: https://github.com/PrestaShop/ADR/blob/master/0017-backward-compatibility-promise.md

I agree with @kpodemski .

What is the benefit of merging this PR versus not merging it? Who does it serve?

**Comment by Hlavtox** [comment]
@kpodemski @matks Ready to go

**Review by kpodemski — DISMISSED** [review]

**Comment by FabienPapet** [comment]
I'm ok with all these changes, but please wait for the https://github.com/PrestaShop/PrestaShop/pull/31138 to be merged because it is a PITA to maintain and it will have conflicts (refacto on form types)

**Review by FabienPapet — CHANGES_REQUESTED** [review]
Putting as request changes just to wait before the refacto on form types. Will rebase myself your PR if needed :D

**Review by FabienPapet — DISMISSED** [review]

**Comment by Hlavtox** [comment]
@FabienPapet Rebased, I chose not to delete TranslateTextType, because some native modules are using it still.

File: classes/Link.php [hunk 1]
```diff
@@ -882,7 +882,7 @@ public function getTabLink(array $tab)
 
     /**
      * Used when you explicitly want to create a LEGACY admin link, this should be deprecated
-     * in 1.8.0.
+     * in 9.x
```
Comment (zuk3975) [inline classes/Link.php:885]: whats the point of having this whole comment if by every version we just move it further to be deprecated in next version?
Either it is deprecated or not. (no "should be deprecated someday")

**Review by zuk3975 — APPROVED** [review]
just a comment that doesn't block

File: classes/Link.php (hunk 1 above)
Comment (kpodemski) [reply classes/Link.php:885]: Let's deprecate it in the next PR 👍🏻

File: src/PrestaShopBundle/Form/Admin/Type/TranslateTextType.php [hunk 2]
```diff
@@ -36,15 +36,15 @@
 /**
  * Class is responsible for creating translatable text inputs.
  *
- * @deprecated since version 1.7.6 and will be removed in 1.8. Use the TranslatableType instead.
+ * @deprecated since version 1.7.6 and will be removed in 9.x, use the TranslatableType instead.
```
Comment (kpodemski) [inline src/PrestaShopBundle/Form/Admin/Type/TranslateTextType.php:39]: and this one should be removed in the next PR :)

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
Auto tests: https://github.com/kpodemski/ga.tests.ui.pr/actions/runs/4222635863

**Review by MhiriFaten — APPROVED** [review]
Hello @Hlavtox ,

LGTM, it is QA :heavy_check_mark:

**Comment by nicosomb** [comment]
Thank you @Hlavtox and @MhiriFaten

**Pull request merged**